### PR TITLE
Hubspot v3 deals and companies

### PR DESCRIPTION
**Pull request by SBurwash**
# What was the issue
Modification to properties + wanted to add deals and companies to tap

# How did we solve it
Added deals and companies. Dynamically query properties from api, and create schema based on properties. No need for recursive behaviour; uri is small enough

# Additional Notes / Warnings
I didn't want to do this, but we'll before a rebase of this branch into the pipeline hubspot_v3_branch first, make sure everything is on the right page

# Fun fact (optional)

### CODE REVIEW

File: tap_hubspot/client.py [hunk 1]
```diff
@@ -121,13 +120,24 @@ def get_json_schema(self, from_type: str) -> dict:
...
             type_name = from_type
         else:
             raise ValueError("Expected `str` or a SQLAlchemy `TypeEngine` object or type.")
-        for sqltype, jsonschema_type in sqltype_lookup.items():
+        for sqltype, jsonschema_type in sqltype_lookup_hubspot.items():
             if sqltype.lower() in type_name.lower():
                 return jsonschema_type
+        # for sqltype, jsonschema_type in sqltype_lookup.items():
+        #     if sqltype.lower() in type_name.lower():
+        #         return jsonschema_type
```
Comment (jeromepl) [inline tap_hubspot/client.py:140]: Can we get rid of this?

File: tap_hubspot/streams.py [hunk 2]
```diff
@@ -35,15 +35,64 @@ class MeetingsStream(HubspotStream):
...
     def get_url_params(self, context: Optional[dict], next_page_token: Optional[Any]) -> Dict[str, Any]:
         params = super().get_url_params(context, next_page_token)
-        params['properties'] = ','.join(self.get_properties_from_file())
+        params['properties'] = ','.join(self.properties)
         return params
 
+    @property
+    def schema(self) -> dict:
+        if self.cached_schema is None:
+            self.cached_schema, self.properties = self.get_custom_schema()
```
Comment (jeromepl) [inline tap_hubspot/streams.py:49]: Did you check if this was needed after all?

File: tap_hubspot/streams.py [hunk 3]
```diff
@@ -35,15 +35,64 @@ class MeetingsStream(HubspotStream):
     path = f"/crm/v3/objects/meetings"
     primary_keys = ["id"]
     replication_key = "updatedAt"
+    cached_schema = None
+    properties = []
```
Comment (jeromepl) [inline tap_hubspot/streams.py:39]: I think this should be on the class `HubspotStream`, not on all subclasses

**Review by jeromepl — APPROVED** [review]
This is nice 👌

File: tap_hubspot/client.py (hunk 1 above)
Comment (SBurwash) [reply tap_hubspot/client.py:140]: Niope, getting rid of it

File: tap_hubspot/streams.py (hunk 2 above)
Comment (SBurwash) [reply tap_hubspot/streams.py:49]: Yes, for some reason it seems we do. Tbh I didn't really want to investigate, just fix

**Pull request merged**